multisense_ros/src/reconfigure.cpp

```diff
@@ -844,20 +941,50 @@ template<class T> void Reconfigure::configureExtrinsics(const T& dyn)
-void Reconfigure::callback_sl_bm_cmv4000_imu  (multisense_ros::sl_bm_cmv4000_imuConfig&  dyn, uint32_t level) { (void) level; SL_BM_IMU();   }
-void Reconfigure::callback_sl_sgm_cmv2000_imu (multisense_ros::sl_sgm_cmv2000_imuConfig& dyn, uint32_t level) { (void) level; SL_SGM_IMU();  }
-void Reconfigure::callback_sl_sgm_cmv4000_imu (multisense_ros::sl_sgm_cmv4000_imuConfig& dyn, uint32_t level) { (void) level; SL_SGM_IMU_CMV4000();  }
-void Reconfigure::callback_mono_cmv2000       (multisense_ros::mono_cmv2000Config&       dyn, uint32_t level) { (void) level; MONO_BM_IMU();   }
-void Reconfigure::callback_mono_cmv4000       (multisense_ros::mono_cmv4000Config&       dyn, uint32_t level) { (void) level; MONO_BM_IMU();   }
-void Reconfigure::callback_s27_AR0234         (multisense_ros::s27_sgm_AR0234Config&     dyn, uint32_t level) { (void) level; S27_SGM();   }
-void Reconfigure::callback_ks21_AR0234        (multisense_ros::ks21_sgm_AR0234Config&    dyn, uint32_t level) { (void) level; KS21_SGM();   }
+// The dynamic reconfigure callbacks (MultiSense S* & feature variations)
+
+void Reconfigure::callback_sl_bm_cmv2000     (multisense_ros::sl_bm_cmv2000Config&      dyn, uint32_t level) { (void) level; SL_BM();              }
+void Reconfigure::callback_sl_bm_cmv2000_imu (multisense_ros::sl_bm_cmv2000_imuConfig&  dyn, uint32_t level) { (void) level; SL_BM_IMU();          }
+void Reconfigure::callback_sl_bm_cmv4000     (multisense_ros::sl_bm_cmv4000Config&      dyn, uint32_t level) { (void) level; SL_BM();              }
+void Reconfigure::callback_sl_bm_cmv4000_imu (multisense_ros::sl_bm_cmv4000_imuConfig&  dyn, uint32_t level) { (void) level; SL_BM_IMU();          }
+void Reconfigure::callback_sl_sgm_cmv2000_imu(multisense_ros::sl_sgm_cmv2000_imuConfig& dyn, uint32_t level) { (void) level; SL_SGM_IMU();         }
+void Reconfigure::callback_sl_sgm_cmv4000_imu(multisense_ros::sl_sgm_cmv4000_imuConfig& dyn, uint32_t level) { (void) level; SL_SGM_IMU_CMV4000(); }
+void Reconfigure::callback_mono_cmv2000      (multisense_ros::mono_cmv2000Config&       dyn, uint32_t level) { (void) level; MONO_BM_IMU();        }
+void Reconfigure::callback_mono_cmv4000      (multisense_ros::mono_cmv4000Config&       dyn, uint32_t level) { (void) level; MONO_BM_IMU();        }
+void Reconfigure::callback_s27_AR0234        (multisense_ros::s27_sgm_AR0234Config&     dyn, uint32_t level) { (void) level; S27_SGM();            }
+void Reconfigure::callback_ks21_AR0234       (multisense_ros::ks21_sgm_AR0234Config&    dyn, uint32_t level) { (void) level; KS21_SGM();           }
+
+void Reconfigure::callback_s27_AR0234_ground_surface        (multisense_ros::s27_sgm_AR0234_ground_surfaceConfig&     dyn, uint32_t level) { (void) level; S27_SGM_GROUND_SURFACE();  }
+void Reconfigure::callback_ks21_AR0234_ground_surface       (multisense_ros::ks21_sgm_AR0234_ground_surfaceConfig&    dyn, uint32_t level) { (void) level; KS21_SGM_GROUND_SURFACE(); }
```

Comment:
I've only added ground surface support for these camera types — I can change this if we want to support different cameras